Submit-PolicyDefinitionFile.ps1

```diff
@@ -0,0 +1,306 @@
+#Requires -PSEdition Core
+
+param(
+    [parameter(Mandatory = $true, Position = 0)] $fileName,
+    [parameter(Mandatory = $false, Position = 1)] [string] $outputDirectory = $null,
+    [switch] $skipFileSplitting
+)
+
+$ErrorActionPreference = "Stop"
+
+$files = Get-ChildItem -Path $fileName
+if ($files.Count -eq 0) {
+    Write-Error "File not found: $fileName."
+}
+elseif ($files.Count -gt 1) {
+    Write-Error "Multiple files found: $fileName. Specify a file, not a directory"
+}
+$file = $files[0]
+$content = Get-Content $file.FullName -Raw
+if (!(Test-Json $content)) {
+    Write-Error "File is not valid JSON: $fileName."
+}
+$definition = ConvertFrom-Json $content -AsHashtable -Depth 100
+
+# analyze structure
+$properties = $definition
+if ($definition.properties) {
+    $properties = $definition.properties
+}
+
+$name = $definition.name
+if (!$name) {
+    if ($definition.id) {
+        $idSplits = $definition.id -split "/"
+        $name = $idSplits[-1]
+    }
+    elseif ($properties.id) {
+        $idSplits = $properties.id -split "/"
+        $name = $idSplits[-1]
+    }
+}
+if ($name) {
+    if (!([guid]::TryParse($name, $([ref][guid]::Empty)))) {
```

Comment:
$name = "" and therefore new GUID is created every time.
type [out] is unknown, if I replace with $([ref][guid]::Empty), it works for me.

Reply:
This was fixed. I will kill the PR and reissue